fido/fido.py

```diff
@@ -24,6 +23,40 @@
 DEFAULT_CONTENT_TYPE = 'application/json'
 
 
+def _url_to_utf8(url):
+    """Makes sure the url is utf-8 encoded"""
+
+    if isinstance(url, unicode):
+        return url.encode('utf-8')
+    return url
+
+
+def _build_body_producer(body, headers):
+    """
+    Prepares the body and the headers for the twisted http request performed
+    by the Twisted Agent.
+
+    :returns: a Twisted FileBodyProducer object as required by Twisted Agent
+    """
+
+    if not body:
+        return None, headers
+
+    bodyProducer = FileBodyProducer(StringIO(body))
+    # content-length needs to be removed because it was computed based on
+    # body but body is now being processed by twisted FileBodyProducer
+    # causing content-length to lose meaning and break the client.
+    # FileBodyProducer will take care of re-computing length and re-adding
+    # a new content-length header later.
+    headers = dict(
+        (key, value)
+        for (key, value) in headers.iteritems()
+        if key != 'Content-Length' and key != 'content-length'
```

Comment:
oops, yep, I had just introduced the .lower() in PR #26.
This is a merge with a past branch of mine ended poorly